Dockerfile

```diff
@@ -1,6 +1,14 @@
+FROM maven:3.5.3-jdk-8-alpine as maven
+
+WORKDIR /app
+COPY ./pom.xml ./pom.xml
+RUN mvn dependency:go-offline -B
+COPY ./src ./src
+RUN mvn package && cp target/petstore-*.jar app.jar
+
 FROM java:8-jre-alpine
 MAINTAINER CyberArk
 
-COPY target/petstore-*.jar /app.jar
+COPY --from=maven /app/app.jar /app.jar
```

Comment:
I'd like see some comments at the top explaining why we're doing this double image and double copying copying process.

Reply:
done :)